addon/-private/system/many-array.js

```diff
@@ -52,9 +54,16 @@ const { get, set } = Ember;
   @uses Ember.MutableArray, Ember.Evented
 */
 export default Ember.Object.extend(Ember.MutableArray, Ember.Evented, {
+  record: null,
+
+  canonicalState: null,
+  currentState: null,
+
+  length: 0,
+
   init() {
     this._super(...arguments);
-
+    this.currentState = Ember.A([]);
```

Comment:
this is a change on master btw - I've merged it into my branch because my branch started so long ago